swarm/src/connection.rs

```diff
@@ -1008,6 +1004,88 @@ mod tests {
         assert!(start.checked_add(duration).is_some())
     }
 
+    #[test]
+    fn compute_new_shutdown_does_not_panic() {
+        let _ = env_logger::try_init();
+
+        #[derive(Clone, Debug)]
+        struct ArbitraryKeepAlive(KeepAlive);
+
+        impl Arbitrary for ArbitraryKeepAlive {
```

Comment:
Couldn't we just implement this on `KeepAlive` directly?

Reply:
The `Arbitrary` implementation and thus `quickcheck` would thus be part of our public API surface, no?